File: Creational/FactoryMethod/C++/products/MacButton.cpp
#include <iostream>
#include "MacButton.h"

/**
 * Concrete product (realisation).
 */

void MacButton::paint()
{
	std::cout << "Painting button in MacOS style." << std::endl;
}

void MacButton::setPosition(const int & x, const int & y)
{
	std::cout << "Setting button position to (" << x << ", " << y << ") in MacOS style." << std::endl;
}

File: Creational/FactoryMethod/Java/products/MacButton.java
package patterns.creational.factorymethod.products;

import patterns.creational.factorymethod.interfaces.IButton;

/**
 * Concrete product.
 */

public class MacButton implements IButton
{
	/**
	 * @Overrride
	 */
	public void paint()
	{
		System.out.println("Painting button in MacOS style.");
	}

	/**
	 * @Override
	 */
	public void setPosition(final int x, final int y)
	{
		System.out.println("Setting button position to (" + x + ", " + y + ") in MacOS style.");
	}
}
